exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java

```diff
@@ -119,6 +119,19 @@ public RawFragmentBatch take() throws IOException, InterruptedException {
       return buffer.take().get();
     }
 
+    @Override
+    public RawFragmentBatch poll(long timeout, TimeUnit timeUnit) throws InterruptedException, IOException {
+      RawFragmentBatchWrapper batchWrapper = buffer.poll(timeout, timeUnit);
+      if (batchWrapper != null) {
+        try {
+          return batchWrapper.get();
+        } catch (InterruptedException e) {
```

Comment:
You should not catch InterruptedException as it is being handled by caller; the handling is also different:
- Upper code is having more logic on how to deal with an interrupted exception
- This code is erasing the interrupted flag whereas caller code doesn't

Reply:
Thanks for catching this. Specifically for spooling buffer implementation even `poll()` method was consuming the InterruptedException. I have made both polls to throw `InterruptedException`.